app/controllers/api/databases.php

```diff
@@ -2535,21 +2535,16 @@ function updateAttribute(
             throw new Exception(Exception::COLLECTION_NOT_FOUND);
         }
 
-        $indexes = $collection->getAttribute('indexes');
-
-        // Search for index
-        $indexIndex = array_search($key, array_map(fn($idx) => $idx['key'], $indexes));
+        $index = $dbForProject->find('indexes', [Query::equal('$id', [$database->getInternalId() . '_' . $collection->getInternalId() . '_' . $key]), Query::limit(1)]);
```

Comment:
Why do we need to make another database query for the index?

Reply:
understood what you mean when using Document->find(). 
I have used the same now.